### PR TITLE
fix: stamp created_by/updated_by/modified_by on remaining ITX proxy writes

### PR DESCRIPTION
**Pull request by jordane**
Resolves [LFXV2-2821](https://linuxfoundation.atlassian.net/browse/LFXV2-2821).

Only the Meeting POST/PUT paths stamped the requesting user on outbound ITX writes; every other create/update from lfx-v2-meeting-service either sent nothing (leaving audit fields blank / attributing changes to the M2M service identity) or, on attachments, sent only `username` without the name/email enrichment the meeting endpoints already do.

## Approach

- Extract `buildRequestingUser` from `MeetingService` into a shared `auditStamper` embedded by every ITX service.
- Wire the existing `UserMetadataReader` (NATS → auth-service) into every service constructor in `main.go`.
- Preserve the graceful-degradation contract: full profile when the reader resolves, `{username, email}` when the reader is nil or errors, and nil when there is no principal on ctx.
- Add sibling `buildRequestingCreatedUpdatedBy` returning the `*itx.CreatedUpdatedBy` shape (no id/avatar) used by attachment endpoints.
- Stamp audit fields **in the service**, never in the converter — so it happens once per call and enrichment lives in one place.

## What now stamps what

| Object | POST | PUT | Notes |
| --- | --- | --- | --- |
| Registrant | `created_by` | `updated_by` | struct fields already existed |
| Meeting occurrence | — | `updated_by` | struct field already existed |
| Past meeting | `created_by` | `updated_by` | added fields to `CreatePastMeetingRequest` |
| Past meeting invitee | `created_by` | `updated_by` | added fields; stamped in Create, update, and the `createInviteeFromUpdate` fallback path |
| Past meeting attendee | `created_by` | `updated_by` | mirror of invitee |
| Past meeting summary | — | `modified_by` | non-standard ITX field name; drops the misleading converter comment claiming ITX derives this from the JWT |
| Meeting attachment (+ presign) | `created_by` w/ name+email | `updated_by` w/ name+email | handlers no longer parse JWT; converters no longer take `username` |
| Past meeting attachment (+ presign) | `created_by` w/ name+email | `updated_by` w/ name+email | mirror of meeting attachment |

Never stamps `created_by` on updates or `updated_by` on creates so we don't overwrite the original creator or leave stale updater data. Presign is treated as a create (ITX persists an attachment record with `upload_status=ongoing` at that point).

## PII redaction in debug logs

`proxy/client.go` already logs full attachment request bodies at `LOG_LEVEL=debug`. Now that `CreatedBy` / `UpdatedBy` on attachments carry `name` and `email`, this PR also:

- Adds `LogValue()` on `itx.User` and `itx.CreatedUpdatedBy` — redacts requester PII to just `username` when either type is logged directly as a `slog` attribute value.
- Adds `requestJSONForLog` in the proxy package — returns a redacted JSON body (audit fields nil'd) for use in the 6 attachment debug-log sites. Wire body sent to ITX is unaffected.

Two layers because `slog` doesn't recursively resolve `LogValuer` on nested struct fields — the `LogValue()` method alone wouldn't fix the pre-existing full-body-as-string debug log pattern.

## Not addressed (out of scope per ticket)

- Past meeting recordings and artifacts (no HTTP create path from this service; ITX populates from webhook/NATS).
- `MeetingsAuditRecord.change_by` (non-standard, not created here).

## Test coverage

Mirrors the existing `meeting_service_test.go` pattern:
- direct `auditStamper` unit tests (including the `(nil, nil)` resolver edge case)
- per-service tests for full-profile stamping, no-principal omission, and create-doesn't-touch-updated / update-doesn't-touch-created
- `TestPastMeetingParticipantService_UpdateParticipant_ResolvesRequesterOnce` regression guard that asserts a single `ResolveProfile` call per participant update across all three scenarios (both-exist, both-missing, mixed)
- redaction tests (`TestRequestJSONForLog_RedactsAuditPII`, `TestUser_LogValue_RedactsPII`, `TestWireBody_StillContainsAuditPII`) that pin down PII stays out of logs and stays in the wire body
- shared helpers (fake reader, no-op ID mapper, ctx builder) extracted to `testutil_test.go`

[LFXV2-2821]: https://linuxfoundation.atlassian.net/browse/LFXV2-2821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/service/itx/past_meeting_participant_service.go [hunk 1]
```diff
@@ -347,6 +368,10 @@ func (s *PastMeetingParticipantService) updateInvitee(
 	pastMeetingID, inviteeID, participantID string,
 	updateReq *itx.UpdateInviteeRequest,
 ) *itx.InviteeResponse {
+	// Stamp updated_by from the authenticated principal so ITX overwrites the stored
+	// value on the invitee record instead of preserving stale data.
+	updateReq.UpdatedBy = s.buildRequestingUser(ctx)
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/past_meeting_participant_service.go]: Good catch — fixed in edd2c07. `UpdateParticipant` now resolves the requesting user once and threads a `*itx.User` through `handleInviteeOperation` / `handleAttendeeOperation` and into the four helpers (`updateInvitee`, `updateAttendee`, `createInviteeFromUpdate`, `createAttendeeFromUpdate`), matching how `CreateParticipant` already does it. Added a `TestPastMeetingParticipantService_UpdateParticipant_ResolvesRequesterOnce` regression guard covering the three scenarios (both-exist, both-missing, mixed) that asserts `reader.calls` is a single-element slice.

File: internal/service/itx/audit.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	user := &itx.User{
+		Username:       principal,
+		Name:           profile.Name,
+		Email:          profile.Email,
+		ProfilePicture: profile.AvatarURL,
+	}
+	if user.Email == "" {
+		user.Email = email
+	}
+	return user
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/audit.go:73]: Fixed in edd2c07. The current `NATSUserMetadataReader` never returns `(nil, nil)` in practice, but the `domain.UserMetadataReader` interface does not forbid it, and a panic here would take down the outbound ITX write — the exact opposite of what the helper is designed to do. Added the nil-profile branch (with the same warn log + `{username, email}` fallback shape as the error path) and a `degrades to username/email when resolver returns (nil, nil)` test case in `audit_test.go`.

File: internal/service/itx/past_meeting_service_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		_, err := svc.UpdatePastMeeting(ctxWithPrincipal("bob", "bob@example.com"), "pm-1", &itx.CreatePastMeetingRequest{
+			ProjectID: "proj-1",
+		})
+		require.NoError(t, err)
+		require.NotNil(t, client.lastUpdateReq.UpdatedBy)
+		assert.Equal(t, "bob", client.lastUpdateReq.UpdatedBy.Username)
+		assert.Equal(t, "bob@example.com", client.lastUpdateReq.UpdatedBy.Email)
+		assert.Nil(t, client.lastUpdateReq.CreatedBy, "update must not overwrite original creator")
+	})
+}
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/past_meeting_service_test.go:83]: Skipping this one — the scenario it guards against does not exist in our request lifecycle.

- `CreatedBy` and `UpdatedBy` are not part of the Goa API surface. Neither the `CreateItxPastMeetingPayload` / `UpdateItxPastMeetingPayload` types nor the registrant equivalents expose them, so clients cannot set either field.
- The `itx.CreatePastMeetingRequest` / `itx.ZoomMeetingRegistrant` structs are constructed fresh from the payload by the converters in `cmd/meeting-api/service/` on every call — there is no reuse or decoded-from-wire path that could carry both fields into the service.
- Only the service itself sets these fields, immediately before dispatch to ITX. The service always sets exactly one, so the other is guaranteed zero-valued (nil pointer) at dispatch.

Adding defensive `req.UpdatedBy = nil` on create / `req.CreatedBy = nil` on update would be guarding against a state we do not own and cannot receive. It would also make the service imperative about the shape of a struct we build ourselves, which invites the opposite bug (accidentally clearing a legitimately-populated field in a future change).

The existing tests already assert the "wrong" field is `nil` after dispatch (see `TestRegistrantService_UpdateRegistrant_StampsUpdatedByNotCreatedBy`, `TestPastMeetingService_UpdatePastMeeting_StampsUpdatedByNotCreatedBy`, and the meeting-service equivalents), so any future converter change that started populating both fields would be caught immediately.

File: internal/service/itx/audit.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+func (a auditStamper) buildRequestingCreatedUpdatedBy(ctx context.Context) *itx.CreatedUpdatedBy {
+	u := a.buildRequestingUser(ctx)
+	if u == nil {
+		return nil
+	}
+	return &itx.CreatedUpdatedBy{
+		Username: u.Username,
+		Email:    u.Email,
+		Name:     u.Name,
+	}
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/audit.go:88]: Good catch — legitimate PII concern. Fixed in the amended [ae24bec](https://github.com/linuxfoundation/lfx-v2-meeting-service/commit/ae24bec) with a two-layer approach:

1. **`LogValue()` on `itx.User` and `itx.CreatedUpdatedBy`** (in `pkg/models/itx/`). When either type is logged directly as a `slog` attribute value (e.g. `slog.Any("created_by", cb)`), only the username is emitted — name/email/avatar are dropped. This is defense in depth for future direct-logging call sites.

2. **`requestJSONForLog(req)` in `internal/infrastructure/proxy/logredact.go`**. This is what actually fixes the reported issue. It shallow-copies known attachment request types, clears the `CreatedBy` / `UpdatedBy` pointer, and marshals the copy — used by the 6 attachment debug-log sites in `client.go`. The original `req` is not mutated, so the wire body sent to ITX still carries the full audit stamp.

The two-layer approach exists because `slog` does **not** recursively resolve `LogValuer` on nested struct fields — so `LogValue()` alone wouldn't catch the pre-existing `"request", string(jsonBody)` pattern where the `User` has already been JSON-marshaled into a string before hitting `slog`.

Test coverage:
- `TestRequestJSONForLog_RedactsAuditPII` — confirms redaction across all 5 attachment request types, and that non-audit request fields (description, filename, etc.) are preserved for debugging.
- `TestRedactAuditForLog_DoesNotMutateOriginal` — guards the wire body against accidental mutation by the log-time redaction.
- `TestWireBody_StillContainsAuditPII` — canonical guard that the actual outbound JSON body still contains name+email so ITX records the correct audit trail.
- `TestUser_LogValue_RedactsPII` / `TestCreatedUpdatedBy_LogValue_RedactsPII` — direct-attribute logging path.

File: go.mod [hunk 5]
```diff
@@ -19,19 +19,19 @@ require (
 	github.com/teambition/rrule-go v1.8.2
 	github.com/vmihailenco/msgpack/v5 v5.4.1
 	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.68.0
-	go.opentelemetry.io/contrib/propagators/jaeger v1.43.0
-	go.opentelemetry.io/otel v1.43.0
+	go.opentelemetry.io/contrib/propagators/jaeger v1.44.0
```
[the remark this answers is not in the document]
Comment (jordane) [reply go.mod:22]: Fair point on scope. Keeping the otel bump on this branch and updated the PR description to call it out explicitly (see the "Additional commit: OpenTelemetry vuln bump" section).

Rationale for keeping it here rather than splitting:
- CI's vulnerability scan was hard-blocking this PR's build on `GO-2026-5158`. Splitting would leave this PR red until an otel-only PR merges, adding a coordination step for a one-line-of-code root cause.
- Same pattern was used in [b50cb84](https://github.com/linuxfoundation/lfx-v2-meeting-service/commit/b50cb84), which bumped `golang.org/x/text` (GO-2026-5970) and `google.golang.org/grpc` (GHSA-hrxh-6v49-42gf) inline with an audit-stamping feature commit — same rationale, same reviewer feedback pattern.
- The bump is isolated to a single self-contained commit ([e1b15ac](https://github.com/linuxfoundation/lfx-v2-meeting-service/commit/e1b15ac)) that can be reverted independently if anything regresses.

`govulncheck ./...` returns clean after the bump. All existing tests pass, including the otel-specific ones (which needed the semconv `v1.40.0` → `v1.41.0` bump to keep the SDK's internal schema and our resource's schema aligned — the SDK started emitting resources against the 1.41.0 schema URL in the new version).

File: internal/infrastructure/proxy/logredact.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+// Only attachment request types currently carry these audit fields on this
+// service's outbound calls (POST/PUT/presign for meeting attachments and
+// past-meeting attachments). Other request types pass through unchanged.
+// The itx.User and itx.CreatedUpdatedBy types also implement slog.LogValuer
+// (see pkg/models/itx), which handles the case where they're logged directly
+// as a slog attribute value; this helper handles the harder case of an audit
+// field nested inside a JSON-encoded request body, which slog cannot resolve
+// on its own.
+func requestJSONForLog(req any) string {
+	b, err := json.Marshal(redactAuditForLog(req))
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/infrastructure/proxy/logredact.go:42]: Good catch — fixed in the amended [c7d0587](https://github.com/linuxfoundation/lfx-v2-meeting-service/commit/c7d0587) with a symmetric response-side helper.

Added `responseJSONForLog(respBody []byte)` in `internal/infrastructure/proxy/logredact.go` and applied it to all 12 attachment response debug-log sites (`client.go:1476, 1529, 1589, 1642, 1702, 1755, 1816, 1863, 1917, 1970, 2031, 2078`). It parses the raw response bytes to `map[string]any`, redacts `created_by` / `updated_by` / `modified_by` / `file_uploaded_by` down to just `{"username": ...}`, and re-marshals. Non-audit response fields (id, name, description, timestamps, URLs, etc.) pass through unchanged so debug logs stay useful.

Covers all three vectors you called out:
1. **Requester echo** on create/update/presign (our own `created_by` / `updated_by` name+email that ITX returns).
2. **Persisted values on GET** — audit stamps from a different user who originally created / last updated the attachment.
3. **`file_uploaded_by`** populated by ITX after upload completion.

Defensive behavior:
- Empty body → empty string (DELETE / 204 path).
- Non-JSON / JSON-array / malformed body → `{"error":"failed to parse response for log"}` marker instead of the raw bytes.
- Audit field present but not object-shaped → value replaced with `null` rather than passed through.

Test coverage (all in `logredact_test.go`):
- `TestResponseJSONForLog_RedactsAuditPII` — parametrized across the 5 audit-shape scenarios (created_by echo, updated_by echo, GET with prior creator+updater, file_uploaded_by, modified_by).
- `TestResponseJSONForLog_UsernameOnly` — username-only shape survives round-trip.
- `TestResponseJSONForLog_EmptyBody` — DELETE / 204 handled.
- `TestResponseJSONForLog_UnparseableBody` — malformed / array / non-JSON produce safe marker, PII in the raw bytes never leaks.
- `TestResponseJSONForLog_UnexpectedAuditShape` — unexpected value shape gets nulled, not forwarded.
- `TestResponseJSONForLog_LeavesNonAuditFieldsAlone` — pins down that debug value is preserved for non-PII fields.

File: internal/service/itx/past_meeting_participant_service.go [hunk 7]
```diff
@@ -88,8 +93,15 @@ func (s *PastMeetingParticipantService) CreateParticipant(
 	var inviteeResp *itx.InviteeResponse
 	var attendeeResp *itx.AttendeeResponse
 
+	// Resolve the requesting user once and stamp created_by on both invitee and
+	// attendee records so the audit trail reflects who added them via the v2 API.
+	creator := s.buildRequestingUser(ctx)
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/past_meeting_participant_service.go:98]: Skipping this one — the no-op case is already rejected before we ever call `buildRequestingUser`.

Lines 88-91 (right above the flagged code):

```go
// Validate that at least one flag is set
if !isInvited && !isAttended {
    return nil, domain.NewValidationError("at least one of is_invited or is_attended must be true")
}
```

That guard fires before the `s.buildRequestingUser(ctx)` call, so the NATS RPC only runs on requests that will actually create at least one record. The claim that "both flags are false ... the API design currently permits" doesn't match the service behavior — a `(false, false)` request gets a validation error, not a silent no-op.

The only remaining edge case where the resolved value would be truly unused is a caller passing `isInvited=true` with a `nil` `inviteeReq` (or the attendee mirror). That's a caller bug — the API handler always builds the request struct whenever the flag is true, so it can't happen through the normal Goa path. Not worth guarding.

File: internal/service/itx/audit.go [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+	"github.com/linuxfoundation/lfx-v2-meeting-service/internal/domain"
+	"github.com/linuxfoundation/lfx-v2-meeting-service/pkg/constants"
+	"github.com/linuxfoundation/lfx-v2-meeting-service/pkg/models/itx"
+)
+
+// auditStamper resolves the requesting principal (stashed on ctx by the JWT auth
+// middleware) into an *itx.User suitable for stamping created_by / updated_by /
+// modified_by on outbound ITX requests. It is embedded by each ITX service so the
+// stamping logic (and its graceful-degradation guarantees) lives in one place.
```
[the remark this answers is not in the document]
Comment (jordane) [reply internal/service/itx/audit.go:18]: Good catch — the docs were out of sync. Fixed in the amended [685ec40](https://github.com/linuxfoundation/lfx-v2-meeting-service/commit/685ec40) by updating all nine affected API contract docs.

Each affected doc now has:
1. A new **`## Audit stamping`** section near the top describing which audit fields the proxy stamps, the shape of the stamped value, and the graceful-degradation behavior when the auth-service lookup fails.
2. Explicit **split** between the Proxy API request body (what clients send) and the ITX API request body (what the proxy forwards). The `"Proxy API & ITX API (Identical)"` label was replaced everywhere the shapes now diverge; the ITX sample JSON was updated to include the stamped audit field.

Files updated:

| Doc | New stamps documented |
| --- | --- |
| `itx-meetings-api.md` | `created_by` (POST) / `updated_by` (PUT) — fixed the pre-existing LFXV2-2809 / LFXV2-2819 gap you called out implicitly |
| `itx-registrants-api.md` | `created_by` (POST) / `updated_by` (PUT) |
| `itx-occurrences-api.md` | `updated_by` (PUT) |
| `itx-past-meetings-api.md` | `created_by` (POST) / `updated_by` (PUT) |
| `itx-past-meeting-invitees-api.md` | `created_by` (POST + PUT-fallback-to-create) / `updated_by` (PUT) |
| `itx-past-meeting-participants-api.md` | `created_by` / `updated_by` on both invitee and attendee paths, incl. the "resolved once, shared across both" guarantee for a single participant PUT |
| `itx-past-meeting-summaries-api.md` | `modified_by` (PUT) — non-standard field name preserved |
| `itx-meeting-attachments-api.md` | `created_by` (POST + presign) / `updated_by` (PUT) — noted the July 2026 `{ username }` → `{ username, name, email }` enrichment |
| `itx-past-meeting-attachments-api.md` | same as meeting-attachments |

Response schemas were left as-is where they still say "Identical" — those are read-only ITX-echoed fields and the shape hasn't changed.

File: internal/infrastructure/proxy/logredact.go [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+// response bodies whose values carry user-identity PII (requester name +
+// email) and must be redacted before appearing in debug logs.
+//
+// Kept as a package-level var so it can be extended if ITX adds new
+// audit-user fields in the future.
+var auditFieldsForResponseRedaction = []string{
+	"created_by",
+	"updated_by",
+	"modified_by",
+	"file_uploaded_by",
```
Comment (dealako) [inline internal/infrastructure/proxy/logredact.go:22]: [minor] Presigned S3 URLs survive redaction and debug is the effective default log level

Issue: `responseJSONForLog` only redacts the keys in `auditFieldsForResponseRedaction` (`created_by`/`updated_by`/`modified_by`/`file_uploaded_by`). The presign and download responses also carry `file_url` / `download_url`, which pass through verbatim.

Proof: The four presign/download sites in `client.go` (1476, 1529, 1589, 1642) log the body via `responseJSONForLog(respBody)`. `MeetingAttachmentPresignResponse.FileURL` / `PastMeetingAttachmentPresignResponse.FileURL` (`pkg/models/itx/attachments.go:45,68`) and `AttachmentDownloadResponse.DownloadURL` (`attachments.go:76`) are all documented "Presigned S3 URL … valid for 60 minutes", and are top-level keys not in the redaction list — so they emit verbatim. `logredact_test.go:285` even pins `"file_url":"…"` as surviving. And this fires by default: `internal/logging/logging.go:26` sets `logLevelDefault = slog.LevelDebug`, used by the `default:` case when `LOG_LEVEL` is unset/unrecognized, so `slog.DebugContext` is live out of the box.

Why it matters: A presigned PUT URL grants unauthenticated write to the attachment blob (~60 min); a GET URL grants unauthenticated read. In a log-aggregation pipeline these are visible to a broader audience than the authorized caller within the validity window.

Fix: Extend the redaction to URL-bearing keys, e.g. add a `file_url` / `download_url` (or any `*_url`) pass in `responseJSONForLog` that replaces the value with `"[redacted-presigned-url]"`, and update the `LeavesNonAuditFieldsAlone` test. Separately worth reconciling: CLAUDE.md says default `LOG_LEVEL` is `info`, but the code defaults to Debug — that mismatch is what makes this on-by-default. Non-blocking / pre-existing; flagging because this PR adds the redaction pass over exactly these bodies.

File: internal/infrastructure/proxy/logredact.go [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+		cp := *r
+		cp.CreatedBy = nil
+		return &cp
+	case *itx.UpdatePastMeetingAttachmentRequest:
+		if r == nil {
+			return req
+		}
+		cp := *r
+		cp.UpdatedBy = nil
+		return &cp
```
Comment (dealako) [inline internal/infrastructure/proxy/logredact.go:90]: [nit] Allow-list redaction has no compile-time guard against a future PII leak

Issue: `redactAuditForLog` is a type-switch allow-list with a `default: return req` pass-through, and the non-attachment client methods are protected only by the fact that they currently don't log bodies. Now that meeting/registrant/past-meeting/summary/participant requests also carry `name`+`email` in their audit fields, a future `slog.DebugContext(..., "request", jsonBody)` added to any of those methods (the pattern this PR just removed from attachments) would silently leak requester PII — and it wouldn't even hit a redaction branch here.

Fix (optional): either extend `redactAuditForLog` to the meeting/registrant/participant/summary request types, or drop a short comment at those non-logging methods noting that any future body logging must route through the redaction helpers. Low likelihood, silent failure mode.

**Review by dealako — APPROVED** [review]
Hi @jordane 👋 — reviewed this after the several bot rounds you've already worked through, and it holds up well.

**Overall:** Clean, well-scoped fix. Extracting `buildRequestingUser` into a shared `auditStamper` and embedding it on every ITX service is the right call — the stamping contract (full profile → `{username,email}` → `nil`) now lives in one documented place instead of being copy-pasted. I traced the create/update separation through every service and it's correct everywhere: creates stamp `created_by` only, updates stamp `updated_by` only (summary uses `modified_by`), the create-from-update fallback correctly stamps `created_by`, and the update structs physically lack a `CreatedBy` field so mis-stamping is impossible by construction. The once-resolve latency optimization in `UpdateParticipant` is correct (shared pointer is read-only, both records get an identical stamp) and pinned by `ResolvesRequesterOnce`. The two-layer PII redaction (`LogValue()` + body helpers) is sound and covers every attachment body-logging sink; the meeting/registrant/past-meeting/summary/participant client methods log no bodies, so their new name/email fields don't leak. Test coverage mirrors the existing pattern and is thorough. `go build ./...` and the `itx` / `proxy` / `models/itx` test packages all pass.

I confirm the prior bot findings you addressed — nil-profile deref guard (edd2c07), request+response redaction (ae24bec/c7d0587), once-resolve latency (edd2c07), and the nine stale contract docs (685ec40) — are all correctly resolved.

One net-new finding the bots didn't surface, plus a couple of non-blocking notes below. None block merge.

**Structured issue count:**
- 🔴 Blocking: 0
- 🟡 Minor: 1 — presigned S3 URLs (`file_url`/`download_url`) pass through the new response-redaction helper unredacted, and debug is the effective default log level (pre-existing; this PR is the natural place to close it)
- ⚪ Nit: 2 — allow-list redaction coupling has no compile-time guard; a future-dated "Prior to July 2026" doc phrasing
- ❔ Question: 0

**Decision: ✅ Approved with minor comments.** The one security-adjacent finding (presigned URLs in debug logs) is pre-existing behavior — these sites already logged `string(respBody)` before this PR — and is arguably out of the ticket's scope, so I'm not blocking on it. But since this PR adds the exact redaction pass over those bodies, it's worth a quick follow-up (here or a separate ticket). Nice work threading the bot feedback.

**Review by andrest50 — APPROVED** [review]
